bcipy/helpers/triggers.py

```diff
@@ -207,6 +214,110 @@ class FlushFrequency(Enum):
     END = "flush at end of session"
 
 
+def read_data(lines: Iterable[str]) -> List[Trigger]:
+    """Read raw trigger data from the given source.
+
+    Parameters
+    ----------
+        data - iterable object where each item is a str with data for a single
+            trigger.
+
+    Returns
+    -------
+        list of all Triggers in the data.
+    """
+    triggers = []
+    for i, line in enumerate(lines):
+        try:
+            trg = Trigger.from_list(line.split())
+            triggers.append(trg)
+        except (AssertionError, ValueError) as trg_error:
+            raise BciPyCoreException(
+                f'Error reading trigger from line {i+1}: {trg_error}'
+            ) from trg_error
+    return triggers
+
+
+def find_starting_offset(triggers: List[Trigger], device_type: str = None) -> Trigger:
+    """Given a list of raw trigger data, determine the starting offset for the
+    given device. The returned trigger has the timestamp of the first sample
+    recorded for the device.
+
+    If no device is provided the EEG offset will be used. If there are
+    no offset triggers in the given data a Trigger with offset of 0.0 will be
+    returned.
+
+    Parameters
+    ----------
+        triggers - list of trigger data; should include Triggers of
+          TriggerType.OFFSET
+        device_type - each device will generally have a different offset. This
+            parameter is used to determine which trigger to use. If not given
+            the EEG offset will be used by default. Ex. 'EYETRACKER'
+    Returns
+    -------
+        The Trigger for the first matching offset for the given device, or a
+            Trigger with offset of 0.0 if a matching offset was not found.
+    """
+    label = 'starting_offset'

```

Comment:
In the current approach, the user must call the TriggerHandler to get the correct timing for each device separately. Assuming the same TriggerTypes are of interest, the other trigger data does not change. This could be reduced to a single call and return a data structure with the device types found as a key to timing with the option to filter as you do now. 

There are a few drawbacks,

- We use this to apply the static offsets, which are different per device. However, this can be moved to the reshapers as all support the static offset application. Alternatively, since the trigger handler is aware of device types already, we could share the devices with their own offsets defined.

- Two calls may be desired to get different TriggerTypes of interest. It's not necessary, though. If the first call contains all labels of interest, the user can filter out what they need or adjust things further, as we often do separating trials.

Reply:
You bring up some good points. Our main use cases so far primarily deal with triggers for one device at a time (ex. training the EEG model, visualizing EEG, training the eye tracker model). My preference is to make the common case easy in the `load` and `trigger_decoder` functions for now.

However, I can see the potential for future scenarios where we might want triggers adjusted for all devices, as you described. Reading through the trigger data multiple times to accomplish this isn't too bad, as this is not a large amount of data, but it's a little clunky.

I added another function for getting a dictionary of offsets by device. This function, along with the other additions in this PR, should provide some flexible building blocks for reading and processing the trigger data as needed. Maybe we can revisit the main APIs after we have more models in place and see if adjusting the APIs would be beneficial.